scripts/compress_images_test.py

```diff
@@ -0,0 +1,261 @@
+        self,
+        mock_stat: mock.MagicMock
+    ) -> None:
+        """Test file sizes are retrieved when actual_compression is False."""
+        mock_stat.return_value.st_size = 10000
+
+        file_path = pathlib.Path('/fake/path/image.png')
+        output_file_path = pathlib.Path('/fake/path/compressed_image.png')
+
+        actual_compression = False
+        if not actual_compression:
+            original_size = file_path.stat().st_size
+            new_size = output_file_path.stat().st_size
+
+        self.assertEqual(original_size, 10000)
+        self.assertEqual(new_size, 10000)
+        mock_stat.assert_any_call()
+
+    def test_run_no_compressible_images(self) -> None:
+        """Test run method when no images are compressible."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        # Here we use object because a single compressible image
+        # dict needs to be wrapped in a list to match the
+        # method's expected return type and testing scenario.
+        with mock.patch.object(
+            compressor, 'find_compressible_images', return_value=[]
+        ):
+
+            result = compressor.run()
+            self.assertEqual(result, 0)
+
+    def test_return_value_based_on_actual_compression(self) -> None:
+        """Test return value based on actual_compression flag."""
+        result_image = [{
+            'path': '/fake/path/image.png',
+            'original_size': 10000,
+            'new_size': 9000
+        }]
+
+        # Case 1.
+        actual_compression = True
+        if actual_compression:
+            result = None
+        else:
+            result = result_image
+        self.assertIsNone(result)
+
+        # Case 2.
+        actual_compression = False
+        if actual_compression:
+            result = None
+        else:
+            result = result_image
+        self.assertEqual(result, result_image)
+
+    @mock.patch('subprocess.run')
+    def test_run_with_compressible_images(
+        self, mock_subprocess_run: mock.MagicMock
+    ) -> None:
+        """Test run method with compressible images."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=os.path.join(self.temp_dir, 'compressed_images')
+        )
+        mock_compressible_images = [
+            {
+                'path': pathlib.Path(self.png_path),
+                'original_size': 1000,
+                'new_size': 500
+            }
+        ]
+        # Here we use object because a single compressible image
+        # dict needs to be wrapped in a list to match the
+        # method's expected return type and testing scenario.
+        with mock.patch.object(
+            compressor,
+            'find_compressible_images',
+            return_value=mock_compressible_images
+        ):
+            result = compressor.run()
+            self.assertEqual(result, 1)
+
+            self.assertEqual(mock_subprocess_run.call_count, 10)
```

Comment:
here and elsewhere, try to avoid having your tests depend on the internal implementation details of the code being tested. Instead, test the overall behavior from a user's perspective. See "Test the interface, not the implementation" here: https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests